From 6488c60f7bb9a8d0f956e51a9f7657db9027c5fe Mon Sep 17 00:00:00 2001
From: Awais Belal <awais_belal@mentor.com>
Date: Wed, 31 Aug 2016 11:36:38 +0500
Subject: [PATCH] Layer-Samples: fix build without BUILD_LAYERS

Some of the layer samples require layer utils to be
built and these fail when BUILD_LAYERS is disabled.
We fix this by providing the required dependencies
even when the BUILD_LAYERS is disabled.

Signed-off-by: Awais Belal <awais_belal@mentor.com>
---
 Layer-Samples/Overlay/CMakeLists.txt | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/Layer-Samples/Overlay/CMakeLists.txt b/Layer-Samples/Overlay/CMakeLists.txt
index 26f9b3a..a26664c 100644
--- a/Layer-Samples/Overlay/CMakeLists.txt
+++ b/Layer-Samples/Overlay/CMakeLists.txt
@@ -40,6 +40,18 @@ add_library(VKLayer_overlay SHARED overlay.cpp
         ${SHADER_DIR}/overlay-frag.spv
         overlay.json)
 
+if(NOT BUILD_LAYERS)
+add_custom_command(OUTPUT vk_dispatch_table_helper.h
+    COMMAND ${PYTHON_CMD} ${PROJECT_SOURCE_DIR}/vk-generate.py ${DisplayServer} dispatch-table-ops layer > ${LAYERS_LOC}/layers/vk_dispatch_table_helper.h
+    DEPENDS ${PROJECT_SOURCE_DIR}/vk-generate.py ${PROJECT_SOURCE_DIR}/vulkan.py)
+
+add_custom_target(generate_vk_layer_helpers DEPENDS
+    vk_dispatch_table_helper.h
+)
+
+add_dependencies(VKLayer_overlay generate_vk_layer_helpers)
+endif()
+
 if (WIN32)
     set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -D_CRT_SECURE_NO_WARNINGS")
     set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D_CRT_SECURE_NO_WARNINGS")
-- 
1.9.1

